nautobot/core/filters.py

```diff
@@ -629,15 +640,28 @@ def _generate_lookup_expression_filters(cls, filter_name, filter_field):
         if field is None:
             return magic_filters
 
+        try:
+            field_type = cls._meta.model._meta.get_field(field_name)
+        except FieldDoesNotExist:
+            field_type = None
```

Comment:
What's the case where we'd hit this exception? Is this for non-model filter-fields (SerializerMethodField, etc.) or something else?

Reply:
```bash


  Registering job nautobot.core.jobs.GitRepositoryDryRun
Traceback (most recent call last):
  File "/opt/nautobot/lib/python3.8/site-packages/django/db/models/options.py", line 608, in get_field
    return self.fields_map[field_name]
KeyError: 'relationship__key'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/nautobot/bin/nautobot-server", line 6, in <module>
    sys.exit(main())
  File "/source/nautobot/core/cli/__init__.py", line 54, in main
    run_app(
  File "/source/nautobot/core/runner/runner.py", line 297, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/opt/nautobot/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/nautobot/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    django.setup()
  File "/opt/nautobot/lib/python3.8/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/opt/nautobot/lib/python3.8/site-packages/django/apps/registry.py", line 122, in populate
    app_config.ready()
  File "/source/nautobot/circuits/apps.py", line 14, in ready
    super().ready()
  File "/source/nautobot/core/apps/__init__.py", line 57, in ready
    register_menu_items(menu_items)
  File "/source/nautobot/core/apps/__init__.py", line 104, in register_menu_items
    item.link = reverse(item.link, args=item.args, kwargs=item.kwargs)
  File "/opt/nautobot/lib/python3.8/site-packages/django/urls/base.py", line 54, in reverse
    app_list = resolver.app_dict[ns]
  File "/opt/nautobot/lib/python3.8/site-packages/django/urls/resolvers.py", line 530, in app_dict
    self._populate()
  File "/opt/nautobot/lib/python3.8/site-packages/django/urls/resolvers.py", line 464, in _populate
    for url_pattern in reversed(self.url_patterns):
  File "/opt/nautobot/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/nautobot/lib/python3.8/site-packages/django/urls/resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/opt/nautobot/lib/python3.8/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/opt/nautobot/lib/python3.8/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/source/nautobot/core/urls.py", line 7, in <module>
    from nautobot.extras.plugins.urls import (
  File "/source/nautobot/extras/plugins/urls.py", line 32, in <module>
    urlpatterns = import_object(f"{plugin_path}.urls.urlpatterns")
  File "/source/nautobot/extras/plugins/utils.py", line 46, in import_object
    spec.loader.exec_module(module)
  File "/source/examples/example_plugin/example_plugin/urls.py", line 6, in <module>
    from nautobot.extras.views import ObjectDynamicGroupsView
  File "/source/nautobot/extras/views.py", line 42, in <module>
    from . import filters, forms, tables
  File "/source/nautobot/extras/filters/__init__.py", line 813, in <module>
    class RelationshipAssociationFilterSet(BaseFilterSet):
  File "/opt/nautobot/lib/python3.8/site-packages/django_filters/filterset.py", line 62, in __new__
    new_class.base_filters = new_class.get_filters()
  File "/source/nautobot/core/filters.py", line 733, in get_filters
    cls._generate_lookup_expression_filters(
  File "/source/nautobot/core/filters.py", line 644, in _generate_lookup_expression_filters
    field_type = cls._meta.model._meta.get_field(field_name)
  File "/opt/nautobot/lib/python3.8/site-packages/django/db/models/options.py", line 610, in get_field
    raise FieldDoesNotExist("%s has no field named '%s'" % (self.object_name, field_name))
django.core.exceptions.FieldDoesNotExist: RelationshipAssociation has no field named 'relationship__key'

```